test/index.js

```diff
@@ -1,8 +1,31 @@
+import assert from 'assert';
 import nock from 'nock';
 import request from 'supertest';
-import slackin from '../lib/index';
+import slackin, { restrictChannels } from '../lib/index';
 
 describe('slackin', () => {
+  describe('.restrictChannels()', () => {
+    it("returns the same list when the query is empty", () => {
+      let results = restrictChannels(['foo', 'bar'], {query: {}});
+      assert.deepEqual(results, ['foo', 'bar']);
+    });
+
+    it("returns the same list when the channel isn't in the list", () => {
+      let results = restrictChannels(['foo', 'bar'], {query: {channel: 'baz'}});
+      assert.deepEqual(results, ['foo', 'bar']);
+    });
+
+    it("returns undefined when there are no channels to restrict", () => {
+      let results = restrictChannels(undefined, {query: {channel: 'foo'}});
+      assert.equal(results, undefined);
+    });
+
+    it("returns undefined when there is an empty list of channels to restrict", () => {
+      let results = restrictChannels([], {query: {channel: 'foo'}});
+      assert.deepEqual(results, []);
+    });
+  });
```

Comment:
These tests may seem overkill, but I was very nervous about creating a security hole where external users could join a channel that wasn't whitelisted through the channel list provided to slackin.